tests/smoke_tests/test_mount_and_storage.py

```diff
@@ -997,6 +1065,53 @@ def test_new_bucket_creation_and_deletion(self, tmp_local_storage_obj,
         out = subprocess.check_output(['sky', 'storage', 'ls'])
         assert tmp_local_storage_obj.name not in out.decode('utf-8')
 
+    @pytest.mark.no_fluidstack
+    @pytest.mark.parametrize('store_type', [
+        pytest.param(storage_lib.StoreType.S3, marks=pytest.mark.aws),
+        pytest.param(storage_lib.StoreType.GCS, marks=pytest.mark.gcp),
+        pytest.param(storage_lib.StoreType.AZURE, marks=pytest.mark.azure),
+        pytest.param(storage_lib.StoreType.IBM, marks=pytest.mark.ibm),
+        pytest.param(storage_lib.StoreType.R2, marks=pytest.mark.cloudflare)
+    ])
+    def test_bucket_sub_path(self, tmp_local_storage_obj_with_sub_path,
+                             store_type):
+        # Creates a new bucket with a local source, uploads files to it
+        # and deletes it.
+        tmp_local_storage_obj_with_sub_path.add_store(store_type)
+
+        # Check files under bucket and filter by prefix
+        files = self.list_all_files(store_type,
+                                    tmp_local_storage_obj_with_sub_path.name)
+        assert len(files) > 0
+        if store_type == storage_lib.StoreType.GCS:
+            assert all([
+                file.startswith(
+                    tmp_local_storage_obj_with_sub_path.name + '/' +
+                    tmp_local_storage_obj_with_sub_path._bucket_sub_path)
+                for file in files
+            ])
+        else:
+            assert all([
+                file.startswith(
+                    tmp_local_storage_obj_with_sub_path._bucket_sub_path)
+                for file in files
+            ])
+
+        # Check bucket is empty, all files under sub directory should be deleted
+        store = tmp_local_storage_obj_with_sub_path.stores[store_type]
+        store._delete_sub_path()
```

Comment:
When I test the delete path for Azure, this exception occurs. However, aws and gcp pass. azure worked before—could it be an account permission problem? @Michaelvll @romilbhardwaj 

```
onseError] This request is not authorized to perform this operation using this permission.
RequestId:0aa67d01-401e-0048-3416-4ef84d000000
Time:2024-12-14T10:55:17.3755800Z
ErrorCode:AuthorizationPermissionMismatch
Content: <?xml version="1.0" encoding="utf-8"?><Error><Code>AuthorizationPermissionMismatch</Code><Message>This request is not authorized to perform this operation using this permission.
RequestId:0aa67d01-401e-0048-3416-4ef84d000000
Time:2024-12-14T10:55:17.3755800Z</Message></Error>
```

Reply:
Hmm, I just checked azure portal and you should have full permissions...

Reply:
OK I will double check the code

Reply:
I need to call `assign_storage_account_iam_role` before calling `delete`